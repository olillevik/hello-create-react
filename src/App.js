import React from 'react';
import './App.css';

import HellowWorldList from './HelloWorldList';

const App = () => {
    return (
        <div className="App">
            <HellowWorldList/>
        </div>);
};

export default App;