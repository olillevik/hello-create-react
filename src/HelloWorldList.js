import React, {Component} from 'react';
import './HelloWorldList.css';

import HelloWorld from './HelloWorld';
import AddGreeter from './AddGreeter';

class HelloWorldList extends Component {

    constructor(props) {
        super(props);
        this.state = {greetings: ['Jim', 'Sally', 'Bender']};
        this.addGreetingToList = this.addGreetingToList.bind(this);
        this.removeGreetingFromList = this.removeGreetingFromList.bind(this);
    }

    render() {
        return (
            <div className='HelloWorldList'>
                <AddGreeter addGreeting={this.addGreetingToList}/>
                {this.renderGreetings()}

            </div>
        );
    }

    renderGreetings() {
        return this.state.greetings.map(name => (
            <HelloWorld key={name} name={name} removeGreeting={this.removeGreetingFromList}/>
        ));
    }

    addGreetingToList(newName) {
        this.setState({greetings: [...this.state.greetings, newName]});
    }

    removeGreetingFromList(removeName) {
        const filteredGreetings = this.state.greetings.filter(name => {
                return name != removeName;
            }
        );
        this.setState({greetings: filteredGreetings});
    }
}

export default HelloWorldList;