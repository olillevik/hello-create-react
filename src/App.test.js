import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import HelloWorldList from "./HelloWorldList";

describe(App, () => {

    const component = shallow(
        <App/>
    );

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders and matches our snapshot', () => {
        const component1 = renderer.create(
            <App/>
        );
        const tree = component1.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has one HelloWorldList', () => {
        expect(component.find(HelloWorldList)).toHaveLength(1);
    });
});
