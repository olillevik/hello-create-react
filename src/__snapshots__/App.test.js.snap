// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`() => {
    return (
        _react2.default.createElement('div', { className: 'App', __source: { fileName: _jsxFileName, lineNumber: 8 }, __self: undefined },
            _react2.default.createElement(_HelloWorldList2.default, { __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: undefined })));

} renders and matches our snapshot 1`] = `
<div
  className="App"
>
  <div
    className="HelloWorldList"
  >
    <div
      className="AddGreeter"
    >
      <input
        onChange={[Function]}
        type="text"
      />
        
      <button
        onClick={[Function]}
      >
        add
      </button>
    </div>
    <div
      className="HelloWorld"
    >
      Hello
       
      Jim
      !
      <br />
      <button
        className="frenchify"
        onClick={[Function]}
      >
        Frenchify!
      </button>
      <br />
      <button
        className="remove"
        onClick={[Function]}
      >
        Remove me!
      </button>
    </div>
    <div
      className="HelloWorld"
    >
      Hello
       
      Sally
      !
      <br />
      <button
        className="frenchify"
        onClick={[Function]}
      >
        Frenchify!
      </button>
      <br />
      <button
        className="remove"
        onClick={[Function]}
      >
        Remove me!
      </button>
    </div>
    <div
      className="HelloWorld"
    >
      Hello
       
      Bender
      !
      <br />
      <button
        className="frenchify"
        onClick={[Function]}
      >
        Frenchify!
      </button>
      <br />
      <button
        className="remove"
        onClick={[Function]}
      >
        Remove me!
      </button>
    </div>
  </div>
</div>
`;
